### ISSUE TITLE
character encoding error from pseudo.py

### ISSUE THREAD

**Issue by scottstoller** [DistAlgo/distalgo#20]
I am using distalgo master branch (1.1.0b12) and python 3.6.4.  when I run

cd da/examples
python -m da.compiler -p ping.da

I get an error trace that ends with the following:
  File "C:\cygwin\home\stoller\public\distalgo\da\compiler\pseudo.py", line 650, in _DomainSpec
    self.write("\u2208")
  File "C:\cygwin\home\stoller\public\distalgo\da\compiler\pseudo.py", line 127, in write
    self.f.write(text)
  File "C:\Program Files\Python36\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2208' in position 0: character maps to <undefined>

scott

**Comment by sadboy** [comment]
Hi Scott,

I can't reproduce this on Linux/Mac, and I don't currently have a Windows machine to test on. Could you check if the following patch fixes the problem? Thanks!

```diff
diff --git a/da/compiler/ui.py b/da/compiler/ui.py
index 36b71d1..37fc575 100644
--- a/da/compiler/ui.py
+++ b/da/compiler/ui.py
@@ -209,7 +209,7 @@ def dafile_to_pseudofile(filename, outname=None, args=None):
     outname = _sanitize_filename(outname)
     daast = daast_from_file(filename, args)
     if daast:
-        with open(outname, "w") as outfd:
+        with open(outname, "w", encoding='utf-8') as outfd:
             DastUnparser(daast, outfd)
             stderr.write("Written pseudo code file %s.\n"% outname)
```

**Comment by scottstoller** [comment]
I think this patch fixes the problem.  the special characters in the generated pseudo-code ("exists", "forall", and "in" symbols) look fine when I open the generated pseudo-code file (e.g., ping.dap) in Notepad or Microsoft Word.  oddly, the special characters appear incorrectly when I open the file in Notepad++ (even though it shows utf-8 as the encoding).  thanks!

**Comment by sadboy** [comment]
This should be fixed in 1.1.0b13, so closing issue.

I don't have experience with Notepad++, but some editors don't explicitly distinguish between the read encoding and the write encoding. So it might be that Notepadd++ is only showing the write encoding (i.e. it's saying that it will save the file using utf-8), but when opening it still uses some heuristics to guess the encoding, which wrongly concludes our generated .dap files to be plain ascii.
